### DISCUSSION TITLE
مشکل در نصب

### DISCUSSION THREAD

**Discussion by erik7667** [evryn/laravel-toman#14] [Q&A] [answered]
سلام . ورژن لاراول من ۷ عه و طبق آموزش اول این رو زدم 
composer require evryn/laravel-toman
بعد اومدم اینو زدم 
php artisan vendor:publish --provider=Evryn\LaravelToman\LaravelTomanServiceProvider --tag=config
بعد این پیام میاد تو کنسول و متاسفانه هیچ فایلی به config  اضافه نمیشه
Unable to locate publishable resources.
مشکل از چیه ؟

**Comment by AmirrezaNasiri** [comment, marked as answer]
سلام،
خروجی دستور زیر رو لطفا ضمیمه میکنین؟
composer require evryn/laravel-toman

ویرایش:
لاراول فایل سرویس پروایدر پکیج رو بوت نمیکنه. دلیلش رو دقیق نمیدونم چون بصورت پیشفرض باید خودش تشخیص بده. یه بار اینو به providers فایل کانفیگ اپ اضافه کنین و تست کنین:
Evryn\LaravelToman\LaravelTomanServiceProvider::class
و یه بار هم composer dump-autoload بزنین.

**Comment by erik7667** [comment]
Using version ^2.0 for evryn/laravel-toman
./composer.json has been updated
Running composer update evryn/laravel-toman
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package fzaninotto/faker is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: evryn/laravel-toman
Discovered Package: facade/ignition
Discovered Package: fideloper/proxy
Discovered Package: fruitcake/laravel-cors
Discovered Package: laravel/tinker
Discovered Package: laravelcollective/html
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Package manifest generated successfully.
66 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

**Comment by erik7667** [comment]
البته من نصب کردم . اپلود هم کردم . اما همچین اروری میده : 
https://nobella.ir/public/pay?order_id=3
میشه نگا کنین ببینید چیه این ارور ؟

**Reply by AmirrezaNasiri** [reply]
روی لاراول 7 تست کنم خبر بدم

**Reply by erik7667** [reply]
ممنون لطف میکنید

**Reply by erik7667** [reply]
جسارتا اگه دسترسی خاصی هم لازمه برای رفع سریع این مورد آی دی تلگرام بدین من هر دسترسیی لازم بود میفرستم خدمتتون . ممنون از پشتیبانی عالیتون

**Reply by AmirrezaNasiri** [reply]
نه نیازی به این اطلاعات نیست. من روی لاراول 7 جدید تست کردم و بدون مشکل بالا اومد.
به نظر مشکل مختص اپ شماست که این پکیج لود نمیشه. محتوای فایل config/app.php چیه؟

**Reply by erik7667** [reply]
`<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Payment Gateway Name
    |--------------------------------------------------------------------------
    |
    | Here you may specify which of the payment gateway drivers below you wish
    | to use as your default payment gateway in order to request payment to be
    | paid and verify incoming webhook callbacks.
    |
    */

    'default' => env('PAYMENT_GATEWAY', 'idpay'),

    /*
    |--------------------------------------------------------------------------
    | Payment Gateways
    |--------------------------------------------------------------------------
    |
    | Here you may configure the payment gateways for your application. Note that
    | each payment gateway driver might have its own configuration set.
    */

    'gateways' => [

        'zarinpal' => [
            // Setting to true makes all payments happen in a testing environment to fake transactions.
            // Set it to false on production to receive real payments.
            'sandbox' => env('ZARINPAL_SANDBOX', false),

            // Merchant ID of your gateway provided by Zarinpal for your gateway
            // only. Looks like this: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
            'merchant_id' => env('ZARINPAL_MERCHANT_ID'),
        ],

        'idpay' => [
            // Setting to true makes all payments happen in a testing environment to fake transactions.
            // Set it to false on production to receive real payments.
            'sandbox' => env('IDPAY_SANDBOX', false),

            // API Key of your gateway provided by IDPay in your dashboard
            // only. Looks like this: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
            'api_key' => env('IDPAY_API_KEY'),
        ],

    ],

    /*
   |--------------------------------------------------------------------------
   | Default Currency
   |--------------------------------------------------------------------------
   |
     Gateways accept different currencies; So in order to avoid confusion
   | when switching between gateways, you can specify your intended currency
   | here, and Toman will convert to proper one automatically.
   | You can of course override it and specify another currency during making
   | requests using Money object too.
   |
   | Supported currencies: "toman", "rial"
   |
   */

    'currency' => 'toman',

    /*
    |--------------------------------------------------------------------------
    | Default Description
    |--------------------------------------------------------------------------
    |
    | Many gateways require a clear description of transaction purpose. You can
    | specify it here globally or simply override it by defining related data()
    | on Payment object.
    | Here, :amount will be replaced by actual payment amount value.
    |
    */

    'description' => 'پرداخت :amount در '.env('APP_NAME', 'اپلیکیشن'),

    /*
    |--------------------------------------------------------------------------
    | Default Callback Route
    |--------------------------------------------------------------------------
    |
    | Here you may specify which named route you want to use as default callback
    | URL. You can always override it with callback() method on Payment object.
    | For example, the route can be defined in routes/web.php:
    |   Route::get('callbacks/payment-verification')->name('payment.callback');
    |
    */

    // 'callback_route' => 'payment.callback',

];
`
خدمت شما

**Reply by AmirrezaNasiri** [reply]
این فایل پابلیش شده کانفیگه توی config/toman.php 🤔 محتوای config/app.php نباید اینجوری باشه.

**Reply by erik7667** [reply]
شرمنده اشتب فرستادم . این خدمت شما 
`<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Application Name
    |--------------------------------------------------------------------------
    |
    | This value is the name of your application. This value is used when the
    | framework needs to place the application's name in a notification or
    | any other location as required by the application or its packages.
    |
    */

    'name' => env('APP_NAME', 'Laravel'),

    /*
    |--------------------------------------------------------------------------
    | Application Environment
    |--------------------------------------------------------------------------
    |
    | This value determines the "environment" your application is currently
    | running in. This may determine how you prefer to configure various
    | services the application utilizes. Set this in your ".env" file.
    |
    */

    'env' => env('APP_ENV', 'production'),

    /*
    |--------------------------------------------------------------------------
    | Application Debug Mode
    |--------------------------------------------------------------------------
    |
    | When your application is in debug mode, detailed error messages with
    | stack traces will be shown on every error that occurs within your
    | application. If disabled, a simple generic error page is shown.
    |
    */

    'debug' => env('APP_DEBUG', false),

    /*
    |--------------------------------------------------------------------------
    | Application URL
    |--------------------------------------------------------------------------
    |
    | This URL is used by the console to properly generate URLs when using
    | the Artisan command line tool. You should set this to the root of
    | your application so that it is used when running Artisan tasks.
    |
    */

    'url' => env('APP_URL', 'http://localhost'),

    'asset_url' => env('ASSET_URL', null),

    /*
    |--------------------------------------------------------------------------
    | Application Timezone
    |--------------------------------------------------------------------------
    |
    | Here you may specify the default timezone for your application, which
    | will be used by the PHP date and date-time functions. We have gone
    | ahead and set this to a sensible default for you out of the box.
    |
    */

    'timezone' => 'UTC',

    /*
    |--------------------------------------------------------------------------
    | Application Locale Configuration
    |--------------------------------------------------------------------------
    |
    | The application locale determines the default locale that will be used
    | by the translation service provider. You are free to set this value
    | to any of the locales which will be supported by the application.
    |
    */

    'locale' => 'en',

    /*
    |--------------------------------------------------------------------------
    | Application Fallback Locale
    |--------------------------------------------------------------------------
    |
    | The fallback locale determines the locale to use when the current one
    | is not available. You may change the value to correspond to any of
    | the language folders that are provided through your application.
    |
    */

    'fallback_locale' => 'en',

    /*
    |--------------------------------------------------------------------------
    | Faker Locale
    |--------------------------------------------------------------------------
    |
    | This locale will be used by the Faker PHP library when generating fake
    | data for your database seeds. For example, this will be used to get
    | localized telephone numbers, street address information and more.
    |
    */

    'faker_locale' => 'en_US',

    /*
    |--------------------------------------------------------------------------
    | Encryption Key
    |--------------------------------------------------------------------------
    |
    | This key is used by the Illuminate encrypter service and should be set
    | to a random, 32 character string, otherwise these encrypted strings
    | will not be safe. Please do this before deploying an application!
    |
    */

    'key' => env('APP_KEY'),

    'cipher' => 'AES-256-CBC',

    /*
    |--------------------------------------------------------------------------
    | Autoloaded Service Providers
    |--------------------------------------------------------------------------
    |
    | The service providers listed here will be automatically loaded on the
    | request to your application. Feel free to add your own services to
    | this array to grant expanded functionality to your applications.
    |
    */

    'providers' => [

        /*
         * Laravel Framework Service Providers...
         */
        Illuminate\Auth\AuthServiceProvider::class,
        Illuminate\Broadcasting\BroadcastServiceProvider::class,
        Illuminate\Bus\BusServiceProvider::class,
        Illuminate\Cache\CacheServiceProvider::class,
        Illuminate\Foundation\Providers\ConsoleSupportServiceProvider::class,
        Illuminate\Cookie\CookieServiceProvider::class,
        Illuminate\Database\DatabaseServiceProvider::class,
        Illuminate\Encryption\EncryptionServiceProvider::class,
        Illuminate\Filesystem\FilesystemServiceProvider::class,
        Illuminate\Foundation\Providers\FoundationServiceProvider::class,
        Illuminate\Hashing\HashServiceProvider::class,
        Illuminate\Mail\MailServiceProvider::class,
        Illuminate\Notifications\NotificationServiceProvider::class,
        Illuminate\Pagination\PaginationServiceProvider::class,
        Illuminate\Pipeline\PipelineServiceProvider::class,
        Illuminate\Queue\QueueServiceProvider::class,
        Illuminate\Redis\RedisServiceProvider::class,
        Illuminate\Auth\Passwords\PasswordResetServiceProvider::class,
        Illuminate\Session\SessionServiceProvider::class,
        Illuminate\Translation\TranslationServiceProvider::class,
        Illuminate\Validation\ValidationServiceProvider::class,
        Illuminate\View\ViewServiceProvider::class,

        /*
         * Package Service Providers...
         */

        /*
         * Application Service Providers...
         */
        App\Providers\AppServiceProvider::class,
        App\Providers\AuthServiceProvider::class,
        // App\Providers\BroadcastServiceProvider::class,
        App\Providers\EventServiceProvider::class,
        App\Providers\RouteServiceProvider::class,
        Collective\Html\HtmlServiceProvider::class,

    ],

    /*
    |--------------------------------------------------------------------------
    | Class Aliases
    |--------------------------------------------------------------------------
    |
    | This array of class aliases will be registered when this application
    | is started. However, feel free to register as many as you wish as
    | the aliases are "lazy" loaded so they don't hinder performance.
    |
    */

    'aliases' => [

        'App' => Illuminate\Support\Facades\App::class,
        'Arr' => Illuminate\Support\Arr::class,
        'Artisan' => Illuminate\Support\Facades\Artisan::class,
        'Auth' => Illuminate\Support\Facades\Auth::class,
        'Blade' => Illuminate\Support\Facades\Blade::class,
        'Broadcast' => Illuminate\Support\Facades\Broadcast::class,
        'Bus' => Illuminate\Support\Facades\Bus::class,
        'Cache' => Illuminate\Support\Facades\Cache::class,
        'Config' => Illuminate\Support\Facades\Config::class,
        'Cookie' => Illuminate\Support\Facades\Cookie::class,
        'Crypt' => Illuminate\Support\Facades\Crypt::class,
        'DB' => Illuminate\Support\Facades\DB::class,
        'Eloquent' => Illuminate\Database\Eloquent\Model::class,
        'Event' => Illuminate\Support\Facades\Event::class,
        'File' => Illuminate\Support\Facades\File::class,
        'Gate' => Illuminate\Support\Facades\Gate::class,
        'Hash' => Illuminate\Support\Facades\Hash::class,
        'Http' => Illuminate\Support\Facades\Http::class,
        'Lang' => Illuminate\Support\Facades\Lang::class,
        'Log' => Illuminate\Support\Facades\Log::class,
        'Mail' => Illuminate\Support\Facades\Mail::class,
        'Notification' => Illuminate\Support\Facades\Notification::class,
        'Password' => Illuminate\Support\Facades\Password::class,
        'Queue' => Illuminate\Support\Facades\Queue::class,
        'Redirect' => Illuminate\Support\Facades\Redirect::class,
        'Redis' => Illuminate\Support\Facades\Redis::class,
        'Request' => Illuminate\Support\Facades\Request::class,
        'Response' => Illuminate\Support\Facades\Response::class,
        'Route' => Illuminate\Support\Facades\Route::class,
        'Schema' => Illuminate\Support\Facades\Schema::class,
        'Session' => Illuminate\Support\Facades\Session::class,
        'Storage' => Illuminate\Support\Facades\Storage::class,
        'Str' => Illuminate\Support\Str::class,
        'URL' => Illuminate\Support\Facades\URL::class,
        'Validator' => Illuminate\Support\Facades\Validator::class,
        'View' => Illuminate\Support\Facades\View::class,
        'Form' => Collective\Html\FormFacade::class,
        'Html' => Collective\Html\HtmlFacade::class,

    ],

];
`

**Reply by AmirrezaNasiri** [reply]
لاراول فایل سرویس پروایدر پکیج رو بوت نمیکنه. دلیلش رو دقیق نمیدونم چون بصورت پیشفرض باید خودش تشخیص بده. یه بار اینو به providers فایل کانفیگ اپ اضافه کنین و تست کنین:
Evryn\LaravelToman\LaravelTomanServiceProvider::class
و یه بار هم composer dump-autoload بزنین.

**Reply by erik7667** [reply]
تشکر . اون ارور لاراول حل شد با این راهنماییتون . اما همچنان به درگاه وصل نمیشه . تو کنترلرم این متد فراخوانی میشه : 
`‍‍`
public function pay(Request $request)
    {

        $order = Order::where('id',$request['order_id'])->first();

        if ($order != null && $order->status == 0) {
            $product = Product::where('id',$order->product_id)->first();
            if ($product != null) {

                if ($product->price_currency == 'toman') {

                    $request1 = Toman::orderId($order->id)
                        ->amount($order->amount)
                        //->callback('https://nobella.ir/public/verify?order_id='.$order->id)
                        ->request();

                    if ($request1->successful()) {
                        // Store created transaction details for verification
                        $transactionId = $request1->transactionId();

                        // Redirect to payment URL
                        return $request1->pay();
                    }

                    if ($request1->failed()) {

                        dd('failed');
                        // Handle transaction request failure.
                    }

                }

            }
        }

    }
    ``
    و خوب نتیجشو میتونید تو لینک زیر ببینید . میوفته تو ایف فیلد 
https://nobella.ir/public/pay?order_id=3

**Reply by AmirrezaNasiri** [reply]
داخل فیلد اینو بذارین ببینین چی خروجی میده. من فکر کنم از کال‌بک باشه که نه توی کانفیگ ست شده و نه اینجا:
```
$request1->throw();
```

**Reply by erik7667** [reply]
همچین متنی میده . میشه راهنمایی کنین چرا ؟ من ادرس کالبک رو هم از کامنت بودن در اودم 
https://nobella.ir/public/pay?order_id=3

**Reply by AmirrezaNasiri** [reply]
اینو باید از پنلشون پیگیری کنین. شاید آدرسی که تو پنلشون وارد کردین با http بوده و اینجا https هست مثلا.

**Reply by erik7667** [reply]
ممنون درست شد

**Reply by AmirrezaNasiri** [reply]
✌✌

**Comment by erik7667** [comment]
jesaratan in error baraye chie ? 

Unable to connect to IDPay endpoint due to unexpected error.

**Reply by erik7667** [reply]
https://nobella.ir/public/pay?order_id=8
in link alan in error ro mide

**Comment by erik7667** [comment]
ya masalan in error baraye chie ? 
Method Evryn\LaravelToman\Gateways\IDPay\CallbackRequest::verify does not exist.

**Reply by AmirrezaNasiri** [reply]
فکر کنم اون CallbackRequest ای که اینجکت کردین، اونی نیست که توی داکیومنت بوده. باید از Evryn\LaravelToman\CallbackRequest استفاده کنین.
